### PR TITLE
fix(graph): keep streamed tool-call argument fragments that are substrings of accumulated args

### LINKED ISSUE #4066: [BUG] 模型流式模式下工具调用（Function Calling）失败：toolInput cannot be null or empty
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

问题描述
在使用模型并开启**流式模式（streaming）**时，工具/函数调用会失败，抛出异常 IllegalArgumentException: toolInput cannot be null or empty。
根因分析 
模型在流式返回工具调用时，会将工具调用信息分块发送：
Chunk 1: {id: "call_xxx", type: "function", name: "prize_search_tool", arguments: ""}
Chunk 2: {id: "call_xxx", type: "function", name: "", arguments: "{\"keyword\":"}
Chunk 3: {id: "call_xxx", type: "function", name: "", arguments: "\"test\""}
Chunk 4: {id: "call_xxx", type: "function", name: "", arguments: "}"}
当前 NodeExecutor.mergeToolCalls() 方法仅根据 ID 进行去重，但没有正确拼接来自多个 chunk 的 name 和 arguments 字段。
导致的结果：
合并后的工具调用可能仍然是 arguments: ""（来自第一个 chunk）
当 
AgentToolNode
 尝试执行工具时，FunctionToolCallback.call() 抛出异常：Assert.hasText(toolInput, "toolInput cannot be null or empty")
异常：
Caused by: java.lang.IllegalArgumentException: toolInput cannot be null or empty
	at org.springframework.util.Assert.hasText(Assert.java:253) ~[spring-core-6.2.12.jar:6.2.12]
	at org.springframework.ai.tool.function.FunctionToolCallback.call(FunctionToolCallback.java:99) ~[spring-ai-model-1.1.2.jar:1.1.2]
	at com.alibaba.cloud.ai.graph.agent.node.AgentToolNode.lambda$executeToolCallWithInterceptors$0(AgentToolNode.java:218) ~[classes/:na]
	at com.alibaba.cloud.ai.graph.agent.node.AgentToolNode.executeToolCallWithInterceptors(AgentToolNode.java:245) ~[classes/:na]
	at com.alibaba.cloud.ai.graph.agent.node.AgentToolNode.apply(AgentToolNode.java:116) ~[classes/:na]
环境信息
Spring AI Alibaba 版本: 1.1.0.0
模型: qwen3-30b-a3b-instruct-2507
模式: 流式模式开启

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Environment

```markdown
Spring AI Alibaba version(s):
```

### Debug logs

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by logicwu0**
Closes #4066

## Problem

When a model streams a tool call, the argument JSON arrives split across chunks and `NodeExecutor.mergeToolArguments()` reassembles it. It short-circuited on `String.contains()`:

```java
if (currentArguments.startsWith(previousArguments) || currentArguments.contains(previousArguments)) return currentArguments;
if (previousArguments.startsWith(currentArguments) || previousArguments.contains(currentArguments)) return previousArguments;
```

An incremental delta can legitimately be a **substring** of the already-accumulated arguments, and the `contains()` branch then drops it. Concrete case: streaming `{"count": 100}` digit-by-digit as `{"count": ` / `1` / `0` / `0` / `}`. When merging the accumulated `{"count": 10` with the final `0`, `previousArguments.contains("0")` is true, so the fragment is dropped and the arguments become `{"count": 10}`.

This is the same class of failure as #4066 (`toolInput cannot be null or empty` / lost tool arguments): the merge already handles the name/args split (commit 5c2a0fb26), but the `contains()` heuristic silently corrupts arguments containing repeated characters.

## Fix

Only a **prefix** (`startsWith`) indicates a cumulative-snapshot chunk (some providers resend the full arguments each chunk) that should replace the accumulator. A mere substring does not. Removing the two `contains()` checks makes incremental fragments always append, while cumulative-snapshot providers are still handled by `startsWith`.

## Tests

Extends `StreamFunctionCallTest` (which already covers the split-chunk merge for #4406):
- `testStreamingRepeatedDigitFragmentsShouldNotBeDropped` — streams `{"count": 100}` digit-by-digit; asserts `{"count": 100}` (fails on pre-fix code: produced `{"count": 10}`).
- `testStreamingCumulativeSnapshotFragmentsShouldNotBeConcatenated` — a provider resending the full arguments each chunk still collapses to a single `{"city":"Beijing"}` (guards the retained `startsWith` path).

All `spring-ai-alibaba-graph-core` executor tests pass; `spotless:apply` run.

## Note on #4066

The original name/arguments split reported in #4066 is already handled by the current `mergeToolCalls` implementation; this PR fixes the remaining argument-corruption edge case and adds regression coverage, so #4066 can be closed.

### CODE REVIEW

**Comment by guanxuc** [comment]
@codex review

File: spring-ai-alibaba-graph-core/src/main/java/com/alibaba/cloud/ai/graph/executor/NodeExecutor.java [hunk 1]
```diff
@@ -477,10 +477,16 @@ private static String mergeToolArguments(String previousArguments, String curren
...
+        // We must only test startsWith (a prefix), NOT contains: an incremental delta can
+        // legitimately be a substring of the already-accumulated arguments (e.g. streaming
+        // {"count": 100} digit-by-digit, where the final "0" is contained in "...10"), and
+        // a contains-based short-circuit would silently drop that fragment.
+        if (currentArguments.startsWith(previousArguments)) {
             return currentArguments;
         }
-        if (previousArguments.startsWith(currentArguments) || previousArguments.contains(currentArguments)) {
+        if (previousArguments.startsWith(currentArguments)) {
             return previousArguments;
```
[the remark this answers is not in the document]
Comment (logicwu0) [reply spring-ai-alibaba-graph-core/src/main/java/com/alibaba/cloud/ai/graph/executor/NodeExecutor.java]: 已在 b7305220b 修复：删除 `previousArguments.startsWith(currentArguments)` 的反向前缀短路，避免把 `{` 这类合法增量片段误判为旧快照；仍保留 `currentArguments.startsWith(previousArguments)` 以兼容不断扩展的累积快照。

新增了 review 中的嵌套对象回归场景：`{"filter":` / `{` / `"x":1}}`，最终合并为 `{"filter":{"x":1}}`。`StreamFunctionCallTest` 4/4 通过，Checkstyle 和 Spotless 通过。

**Review by oss-sentinel-ai — COMMENTED** [review]
### Summary
Reviewed the changes in this PR (169 lines changed). The changes appear to be well-structured.

### Observations
- Changes span 169 lines
- Please ensure all edge cases are covered
- Verify test coverage for the modified code paths

---
*Automated review by github-manager-bot*
